lib/OA/Dal/Delivery/mysqli.php

```diff
@@ -0,0 +1,217 @@
+    $dbUser     = $dbConf['username'];
+    $dbPassword = $dbConf['password'];
+    $dbName     = $dbConf['name'];
+    if ($dbConf['persistent']) {
+        $dbLink = mysqli_connect('p:' . $dbHost, $dbUser, $dbPassword);
+    } else {
+        $dbLink = mysqli_connect($dbHost, $dbUser, $dbPassword);
+    }
+    if (mysqli_select_db($dbLink, $dbName)) {
+        if (!empty($dbConf['mysql4_compatibility'])) {
+            mysqli_query($dbLink, "SET SESSION sql_mode='MYSQL40'");
+        }
+        if (!empty($conf['databaseCharset']['checkComplete']) && !empty($conf['databaseCharset']['clientCharset'])) {
+            mysqli_query($dbLink, "SET NAMES '{$conf['databaseCharset']['clientCharset']}'");
+        }
+
+        return $dbLink;
+    }
+    OX_Delivery_logMessage('DB connection error: ' . mysqli_error($dbLink), 4);
+    return false;
+}
+
+/**
+ * The function to pass a query to a database link
+ *
+ * @param string $query    The SQL query to execute
+ * @param string $database The database to use for this query
+ *                         (Must match the database section name in the conf file)
+ * @return resource|false  The MySQL resource if the query suceeded
+ *                          or false on failure
+ */
+function OA_Dal_Delivery_query($query, $database = 'database') {
+    // Connect to the database if necessary
+    $dbName = ($database == 'rawDatabase') ? 'RAW_DB_LINK' : 'ADMIN_DB_LINK';
+
+    if (empty($GLOBALS['_MAX'][$dbName])) {
+        $GLOBALS['_MAX'][$dbName] = OA_Dal_Delivery_connect($database);
+    }
+    if ($GLOBALS['_MAX'][$dbName] instanceof mysqli) {
+        $result = mysqli_query($GLOBALS['_MAX'][$dbName], $query);
+        if (!$result) {
+            OX_Delivery_logMessage('DB query error: ' . mysqli_error($GLOBALS['_MAX'][$dbName]), 4);
+            OX_Delivery_logMessage(' - failing query: ' . $query, 5);
+        }
+        return $result;
+    } else {
+        return false;
+    }
+}
+
+/**
+ * The function to fetch a result from a database resource
+ *
+ * @param mysqli_result $resource The MySQL resource
+ * @return array
+ */
+function OA_Dal_Delivery_fetchAssoc($resource) {
+    return mysqli_fetch_assoc($resource);
+}
+
+/**
+ * The function to retrieve the last-insert-id from the database
+ *
+ * @param string $database The name of the database config to use
+ *                         (Must match the database section name in the conf file)
+ * @param string $table    The name of the table we need to get the ID from
+ * @param string $column   The name of the column we need to get the ID from
+ * @return int|false       The last insert ID (zero if last query didn't generate an ID)
+ *                         or false on failure
+ */
+function OA_Dal_Delivery_insertId($database = 'database', $table = '', $column = '')
+{
+    $dbName = ($database == 'rawDatabase') ? 'RAW_DB_LINK' : 'ADMIN_DB_LINK';
+    if (!isset($GLOBALS['_MAX'][$dbName]) || !($GLOBALS['_MAX'][$dbName] instanceof mysqli)) {
```

Comment:
Can you please rewrite this to use your own added function `OA_Dal_Delivery_isValidResult()`? It is basically the same code, if you handle `$GLOBALS['_MAX'][$dbName]` over:
````
if (!isset($GLOBALS['_MAX'][$dbName]) || !(OA_Dal_Delivery_isValidResult($GLOBALS['_MAX'][$dbName]))) {
````